### DISCUSSION TITLE
Can't connect to smart contract using JavaScript!

### DISCUSSION THREAD

**Discussion by workmina140** [WalletConnect/walletconnect-monorepo#2853] [Developer Support]
ываываы

**Comment by chris13524** [comment]
Hey @workmina140 ! Thanks for reporting this. Couple things:
- Can you test this on the latest versions of `@web3modal/ethereum` and `@web3modal/html` which is 2.6.1
- What is the error message you are getting?
- What wallet are you testing with?

**Reply by workmina140** [reply]
main.js?v=40603981446:73 Uncaught ReferenceError: readContract is not defined
metamask
